apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,694 @@
+      case task(Category, Id) of
+        undefined ->
+          {next_state, tasks, State#state{pools = NewPools}};
+        Task ->
+          {NextState, NewState} = choose_pool(State#state{retasks = fyler_queue:in(Task, ReTasks), pools = NewPools}),
+          {next_state, NextState, NewState}
+      end;
+    _ ->
+      {next_state, tasks, State#state{pools = Pools}}
+  end;
+
+tasks_({run, #task{priority = Priority} = Task}, #state{tasks = Tasks, priorities = Priorities} = State) ->
+  update_ets(Task),
+  NumPriority = maps:get(Priority, Priorities, 1),
+  {next_state, tasks, State#state{tasks = fyler_queue:in(Task, NumPriority, Tasks)}};
+
+tasks_({cancel_task, Id}, #state{category = Category} = State) ->
+  case ets:lookup(Category, Id) of
+    [#current_task{pool = Node, status = progress}] ->
+      ets:delete(Category, Id),
+      rpc:cast(Node, fyler_pool, cancel_task, [Id]);
+    [#current_task{status = queued}] ->
+      ets:delete(Category, Id);
+    _ ->
+      ok
+  end,
+  {next_state, tasks, State}.
+
+handle_event(_Event, StateName, State) ->
+  {next_state, StateName, State}.
+
+handle_sync_event(_Event, _From, StateName, State) ->
+  {reply, ok, StateName, State}.
+
+handle_info(start, start, State) ->
+  start(start, State);
+
+handle_info(zero, start, State) ->
+  start(start, State);
+
+handle_info({check, Ref}, zero, State) ->
+  zero({check, Ref}, State);
+
+handle_info({check, Ref}, pools, State) ->
+  pools({check, Ref}, State);
+
+handle_info({check, Ref}, tasks, State) ->
+  tasks({check, Ref}, State);
+
+handle_info(_Info, StateName, State) ->
+  {next_state, StateName, State}.
+
+terminate(_Reason, _StateName, _State) ->
+  ok.
+
+code_change(_OldVsn, StateName, State, _Extra) ->
+  {ok, StateName, State}.
+
+%% ------------------------------------------------------------------
+%% Internal Function Definitions
+%% ------------------------------------------------------------------
+
+apply_state(start, Event, State) ->
+  start(Event, State);
+
+apply_state(zero, Event, State) ->
+  zero(Event, State);
+
+apply_state(pools, Event, State) ->
+  pools(Event, State);
+
+apply_state(tasks, Event, State) ->
+  tasks(Event, State).
+
+%% @doc Choose pool
+
+choose_pool(#state{retasks = ReTasks, tasks = Tasks, pools = Pools} = State) ->
+  {NextState, [NewReTasks, NewTasks], NewPools} = choose_pool([ReTasks, Tasks], Pools, [], []),
+  {NextState, State#state{retasks = NewReTasks, tasks = NewTasks, pools = NewPools}}.
+
+choose_pool([Queue|Queues], [{Node, enabled}|Pools], QueueAcc, PoolsAcc) ->
+  case fyler_queue:out(Queue) of
+    {empty, Queue} ->
+      choose_pool(Queues, [{Node, enabled}|Pools], [Queue|QueueAcc], PoolsAcc);
+    {{value, Task}, NewQueue} ->
+      case send_to_pool(Node, Task) of
+        {Ref, Id} -> choose_pool([NewQueue|Queues], Pools, QueueAcc, [{Node, {pending, Ref, Id}}|PoolsAcc]);
+        _ -> choose_pool([NewQueue|Queues], [{Node, enabled}|Pools], QueueAcc, PoolsAcc)
```

Comment:
Line 496 is too long:         _ -> choose_pool([NewQueue|Queues], [{Node, enabled}|Pools], QueueAcc, PoolsAcc).